libs/platform/src/lib/components/wizard-generator/base-wizard-generator.ts

```diff
@@ -0,0 +1,314 @@
+        return this._visibleItems || this._items;
+    }
+
+    set visibleItems(items: WizardGeneratorItem[]) {
+        this._visibleItems = items;
+    }
+
+    /**
+     * @description Whether or not Wizard is currently on the first step
+     */
+    get isFirstStep(): boolean {
+        return this.visibleItems[0]?.status === 'current';
+    }
+
+    /**
+     * @description Whether or not Wizard is currently on the last step
+     */
+    get isLastStep(): boolean {
+        const lastStep = this.visibleItems[this.visibleItems.length - 1]
+        return lastStep?.status === 'current' && !lastStep.branching;
+    }
+
+    get isBranchingStep(): boolean {
+        const currentIndex = this._wizardGeneratorService.getCurrentStepIndex();
+
+        return this._visibleItems[currentIndex].branching;
+    }
+
+    /**
+     * @hidden
+     */
+    _visibleItems: WizardGeneratorItem[];
+
+    /**
+     * @hidden
+     */
+    private _items: WizardGeneratorItem[];
+
+    /**
+     * @hidden
+     */
+    private _formPending = false;
+
+    /**
+     * @hidden
+     */
+    protected _allowSubscribe = true;
+
+    /**
+     * @hidden
+     */
+    private _navigationButtonLabels = DEFAULT_WIZARD_NAVIGATION_BUTTONS;
+
+    constructor(
+        protected _wizardGeneratorService: WizardGeneratorService,
+        private _cd: ChangeDetectorRef
+    ) {
+        this._wizardGeneratorService.getVisibleSteps()
+        .pipe(takeWhile(() => this._allowSubscribe), debounceTime(10))
+        .subscribe((visibleSteps) => {
+            this.visibleItems = visibleSteps;
+            this._cd.detectChanges();
+        });
+
+        this._wizardGeneratorService.trackStepsComponents()
+        .pipe(takeWhile(() => this._allowSubscribe), debounceTime(10))
+        .subscribe(async (stepsComponents) => {
+            if (stepsComponents.size === this.items?.length) {
+                await this._setVisibleSteps();
+            }
+        });
+    }
+
+    /**
+     * @hidden
+     */
+    ngOnDestroy(): void {
+        this._allowSubscribe = false;
+    }
+
+    /**
+     * @description Reverts Wizard progress one step back.
+     */
+    goBack(): void {
+        const currentStepIndex = this._wizardGeneratorService.getCurrentStepIndex();
+
+        if (this.visibleItems[currentStepIndex - 1] !== undefined) {
+            this.visibleItems[currentStepIndex].status = 'upcoming';
+            this.visibleItems[currentStepIndex - 1].status = 'current';
+        }
+
+        this._wizardGeneratorService.setVisibleSteps(this.visibleItems);
+
+        this._cd.detectChanges();
+    }
+
+    /**
+     * @description Progresses the Wizard one step further.
+     */
+    async goNext(): Promise<void> {
+
+        if (this._formPending) {
+            return;
+        }
+
+        const currentStepIndex = this._wizardGeneratorService.getCurrentStepIndex();
+
+        this._wizardGeneratorService.validateStepForms()
+        .subscribe(async (result) => {
```

Comment:
Do you unsubscribe from this?

Reply:
submitStepForms have take(1) pipe which unsubscribes automatically after first result